gatherers/service-worker.js

```diff
@@ -20,11 +20,15 @@ const Gather = require('./gather');
 
 class ServiceWorker extends Gather {
   setup(options) {
+    this.resolved = false;
+
     this.artifactsResolved = new Promise((res, _) => {
       options.driver.on(
           'ServiceWorker.workerVersionUpdated', data => {
-            if (ServiceWorker.getActivatedServiceWorker(data.versions) !== undefined) {
+            if (ServiceWorker.getActivatedServiceWorker(data.versions) !== undefined &&
+                !this.resolved) {
               this.artifact = {serviceWorkers: data};
+              this.resolved = true;
```

Comment:
Why the check here? I get that it stops you resolving multiple times, but it seems unnecessary given you can't resolve a Promise multiple times.

If it's to stop the artifact being overwritten then that's the only statement that should be inside the if clause.